### PR TITLE
[k8s-keystone-auth] Support role mapping

### PR DESCRIPTION
**Pull request by lingxiankong**
**The binaries affected**:

- [ ] openstack-cloud-controller-manager
- [ ] cinder-csi-plugin
- [x] k8s-keystone-auth
- [ ] client-keystone-auth
- [ ] octavia-ingress-controller
- [ ] manila-csi-plugin
- [ ] manila-provisioner
- [ ] magnum-auto-healer
- [ ] barbican-kms-plugin

**What this PR does / why we need it**:
Support conversion from Keystone roles to Kubernetes users or groups during authentication, so that the Kubernetes cluster admin could config RBAC based on Keystone roles. Please see more details in the doc.

**Which issue this PR fixes**:
fixes #

**Special notes for reviewers**:
Other changes in this PR:

- Fetch user's groups from Keystone.
- Use kebab-case for the config options
- Fix the import alias
- Rename pkg/identity/keystone/syncconfig.yaml to pkg/identity/keystone/sync_test.yaml because it's only used for UT.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/util/util.go [hunk 1]
```diff
@@ -25,3 +25,13 @@ func StringListEqual(list1, list2 []string) bool {
 
 	return s1.Equal(s2)
 }
+
+// Contains searches if a string list contains the given string or not.
+func Contains(list []string, strToSearch string) bool {
```
Comment (kayrus) [inline pkg/util/util.go]: there is already a defined func with the same functionality:
https://github.com/kubernetes/cloud-provider-openstack/blob/d38032ff1a7da28a0648d8f6e3ac5d3750f1a53f/pkg/autohealing/utils/utils.go#L20

Comment (lingxiankong) [reply pkg/util/util.go]: Thanks @kayrus, the plan is to move the common utility functions to a single place gradually. That said, we should remove `Contains` function for autohealing and use this one instead, but that's another PR.

**Comment by lingxiankong** [comment]
The job cloud-provider-openstack-acceptance-test-k8s-cinder has problem that is going to be fixed in https://github.com/theopenlab/openlab-zuul-jobs/pull/809

**Comment by lingxiankong** [comment]
/override cloud-provider-openstack-acceptance-test-k8s-cinder

just a test

**Comment by lingxiankong** [comment]
/override openlab/cloud-provider-openstack-acceptance-test-k8s-cinder

This patch has nothing to do with in-tree cinder, so override the job here.

File: go.sum [hunk 2]
```diff
@@ -545,6 +545,8 @@ github.com/stretchr/testify v1.2.2/go.mod h1:a8OnRcib4nhh0OaRAV+Yts87kKdq0PP7pXf
 github.com/stretchr/testify v1.3.0/go.mod h1:M5WIy9Dh21IEIfnGCwXGc5bZfKNJtfHm1UVUgZn+9EI=
 github.com/stretchr/testify v1.4.0 h1:2E4SXV/wtOkTonXsotYi4li6zVWxYlZuYNCXe9XRJyk=
```
Comment (ramineni) [inline go.sum]: @lingxiankong looks like old dependencies still exist, please run go mod tidy to cleanup the same.

Comment (lingxiankong) [reply go.sum]: weird, I scrolled up a little bit, found there are many modules that have different versions, did you happen to know the reason?

Comment (ramineni) [reply go.sum]: ya, strange , I thought go mod tidy should do required cleanup of the older versions?
One reason may be older versions are some indirect dependency of some package, but not sure

**Comment by ramineni** [comment]
cc @chrigl

File: go.sum (hunk 2 above)
Comment (kayrus) [reply go.sum]: it is only a hashsum, not a direct dependency. if you remove go.sum and run `go mod tidy` again, it should cleanup old dep hashes.

Comment (lingxiankong) [reply go.sum]: @kayrus Do you mean we need to remove go.sum file every time before modifying go.mod in order to remove those hashsum? Does the hashsum bring any effect to the build process?

Comment (kayrus) [reply go.sum]: > Does the hashsum bring any effect to the build process?
no
> Do you mean we need to remove go.sum file every time before modifying go.mod in order to remove those hashsum?
I'd recommend to ignore this next time

**Pull request merged**